runtime/runtime/src/state_viewer/mod.rs

```diff
@@ -264,12 +264,13 @@ impl TrieViewer {
             Err(errors::CallFunctionError::VMError { error_message: message })
         } else {
             let outcome = outcome.unwrap();
-            debug!(target: "runtime", "(exec time {}) result of execution: {:#?}", time_str, outcome);
-            logs.extend(outcome.logs);
-            let mut result = vec![];
-            if let ReturnData::Value(buf) = &outcome.return_data {
-                result = buf.clone();
-            }
+            debug!(target: "runtime", "(exec time {}) result of execution: {}", time_str, outcome);
+            let VMOutcome { logs: outcome_logs, return_data, .. } = outcome;
+            logs.extend(outcome_logs);
+            let result = match return_data {
```

Comment:
```suggestion
            logs.extend(outcome.logs);
            let result = match outcome.return_data {
```

Seems like there's no need to destruct here?

Reply:
interesting. I didn't realize that the borrow checker is smart enough to understand that :)

Reply:
Yeah, borrow checker understands partial moves and partial borrows. What it doesn't understand is partial closure captures:

```rust
let outcome = ...
let closure = || outcome.logs.drain(...);
// The whole outcome is borrowed here, you can't use `.return_data`
```

Though, even this will be fixed later next year.